internal/engine/integration/procedure_test.go

```diff
@@ -206,56 +163,198 @@ func Test_Procedures(t *testing.T) {
+	type testcase struct {
+		name string
+		// foreign is the foreign procedure definition.
+		// It will be deployed in a separate schema.
+		foreign string
+		// otherProc is the procedure that calls the foreign procedure.
+		// It will be included with the foreign procedure.
+		// It should be formattable to allow the caller to format with
+		// the target dbid, and the target procedure should be hardcoded.
+		otherProc string
+		// inputs are the inputs to the test procedure.
+		inputs []any
+		// outputs are the expected outputs from the test procedure.
+		outputs [][]any
+		// if wantErr is not empty, the test will expect an error containing this string.
+		// We use a string, instead go Go's error type, because we are reading errors raised
+		// from Postgres, which are strings.
+		wantErr string
+	}
+
+	tests := []testcase{
+		{
+			name:    "foreign procedure takes nothing, returns nothing",
+			foreign: `foreign procedure do_something()`,
+			otherProc: `procedure call_foreign() public {
+				do_something['%s', 'delete_users']();
+			}`,
+		},
+		{
+			name:    "foreign procedure takes nothing, returns table",
+			foreign: `foreign procedure get_users() returns table(id uuid, name text, wallet_address text)`,
+			otherProc: `procedure call_foreign() public returns table(username text) {
+				return select name as username from get_users['%s', 'get_users']();
+			}`,
+			outputs: [][]any{
+				{"satoshi"},
+				{"wendys_drive_through_lady"},
+				{"zeus"},
+			},
+		},
+		{
+			name:    "foreign procedure takes values, returns values",
+			foreign: `foreign procedure id_from_name($name text) returns (id uuid)`,
+			otherProc: `procedure call_foreign($name text) public returns (id uuid) {
+				return id_from_name['%s', 'id_from_name']($name);
+			}`,
+			inputs:  []any{"satoshi"},
+			outputs: [][]any{{satoshisUUID}},
+		},
+		{
+			name:    "foreign procedure expects args, receives none",
+			foreign: `foreign procedure id_from_name() returns (id uuid)`,
+			otherProc: `procedure call_foreign() public returns (id uuid) {
+				return id_from_name['%s', 'id_from_name']();
+			}`,
+			wantErr: `requires 1 arg(s)`,
+		},
+		{
+			name:    "foreign procedure expects args, receives none",
+			foreign: `foreign procedure get_users($name text) returns table(id uuid, name text, wallet_address text)`,
+			otherProc: `procedure call_foreign() public returns table(username text) {
+				return select name as username from get_users['%s', 'get_users']('satoshi');
+			}`,
+			wantErr: "requires no args",
+		},
+		{
+			name:    "foreign procedure expects 1 arg, receives 2",
+			foreign: `foreign procedure id_from_name($name text, $name2 text) returns (id uuid)`,
+			otherProc: `procedure call_foreign() public returns (id uuid) {
+				return id_from_name['%s', 'id_from_name']('satoshi', 'zeus');
+			}`,
+			wantErr: "requires 1 arg(s)",
+		},
+		{
+			name:    "foreign procedure returns 1 arg, implementation returns none",
```

Comment:
As you pointed out, the test name since here are less confusing. It would be better we change above test case name